Managers/SceneManager.gd

```diff
@@ -75,3 +79,38 @@ func on_defeat() -> void:
 ## Give rewards, allow player to move on the map
 func on_event_win() -> void:
 	PlayerManager.player_room.room_event.on_event_ended()
+
+func _save_scene_data() -> void:
+	for children: Node in current_scene.get_children():
+		children.owner = current_scene
+	var packed_scene: PackedScene = PackedScene.new()
+	packed_scene.pack(current_scene)
+	ResourceSaver.save(packed_scene, "user://current_scene.tscn")
+	
+	var config_file: ConfigFile = SaveManager.config_file
+	config_file.set_value("SceneManager", "current_event", current_event)
+	
+	var error: Error = config_file.save("user://save_data.ini")
+	if error:
+		print("Error saving player data: ", error)
+
+func load_scene_data() -> void:
+	call_deferred("_deferred_load_current_scene_from_data")
+
+func _deferred_load_current_scene_from_data() -> void:
+	var config_file: ConfigFile = SaveManager.load_config_file()
+	if config_file == null:
+		return
```

Comment:
should there be a warning for this ? is it possible to normally call this with a null save file ?

Reply:
It technically isn't where we are calling it from we already do a check on config_file. I'll add a warning for it.